include/ebpf_extension.h

```diff
@@ -91,18 +94,44 @@ typedef struct _ebpf_extension_program_dispatch_table
     ebpf_program_batch_end_invoke_function_t ebpf_program_batch_end_invoke_function;
 } ebpf_extension_program_dispatch_table_t;
 
-typedef struct _ebpf_extension_data
+typedef struct _ebpf_extension_data_v0
 {
     ebpf_extension_header_t header;
     const void* data;
-} ebpf_extension_data_t;
+} ebpf_extension_data_v0_t;
+
+typedef struct _ebpf_extension_data_v1
+{
+    ebpf_extension_header_t header;
+    union
+    {
+        uint64_t as_uint64;
+        struct
+        {
+            bool prog_attach_flags : 1; ///< Program attach flags are supported.
+        };
+    } capabilities;
+    const void* data;
```

Comment:
What's the rationale for adding this in the middle of v0 fields instead of appending to the end as is usually done?

Reply:
This was originally for testing (to force code that didn't correctly check the version to break). Reverted.